### PR TITLE
Add unit tests for geodesic regression on SE2

### PR DESCRIPTION
**Pull request by ninamiolane**
This PR adds unit tests for geodesic regression on SE2, with the backends tf and autograd (torch is missing the implementation of the matrix expm?).

### CODE REVIEW

**Comment by nguigs** [comment]
> This PR adds unit tests for geodesic regression on SE2, with the backends tf and autograd (torch is missing the implementation of the matrix expm?).

PyTorch does have expm but doesn't have logm. Apparently the problem comes from the implementation on GPU, but we could use [this](https://github.com/pytorch/pytorch/issues/9983#issuecomment-891777620) workaround to have a differentiable version of logm in pytorch?

File: geomstats/learning/geodesic_regression.py [hunk 1]
```diff
@@ -238,6 +257,53 @@ def _fit_extrinsic(self, X, y, weights=None, compute_training_score=False):
 
         return self
 
+    def initialize_parameters(self, y):
```
Comment (ninamiolane) [inline geomstats/learning/geodesic_regression.py:260]: Nice!! 🎉

File: geomstats/_backend/numpy/autodiff.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def detach(x):
     return x
 
 
-def value_and_grad(func):
+def value_and_grad(*args, **kwargs):
```
Comment (ninamiolane) [inline geomstats/_backend/numpy/autodiff.py:21]: `value_and_grad(func, to_numpy=False)` could work too and would more closely follow the API of the others.

File: tests/tests_geomstats/test_examples.py [hunk 3]
```diff
@@ -68,6 +69,11 @@ def test_empirical_frechet_mean_uncertainty_sn():
     def test_geodesic_regression_hypersphere():
         geodesic_regression_hypersphere.main()
 
+    @staticmethod
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_examples.py:72]: Nice!!

**Review by ninamiolane — COMMENTED** [review]
Wohoooo! It's happening, congrats on making it work 🎉 

I let you merge when you feel like it. (Maybe keep the grassmanian regression for the next PR?)

File: geomstats/_backend/numpy/autodiff.py (hunk 2 above)
Comment (nguigs) [reply geomstats/_backend/numpy/autodiff.py:21]: The point is to get the wrong backend error instead of a possible error due to the arguments

**Pull request merged**